04_numeronym/shahriys/main_test.go

```diff
@@ -0,0 +1,56 @@
+package main
+
+import (
+	"bytes"
+	"reflect"
+	"testing"
+)
+
+func TestMain(t *testing.T) {
+	var bufnum bytes.Buffer
+	outnum = &bufnum
+
+	main()
+	expected := `[a11y K8s abc]`
+	actual := bufnum.String()
+
+	if expected != actual {
+		t.Errorf("Unexpected output in main()" + expected + " " + actual)
+	}
+
+}
+func TestLettersSingle(t *testing.T) {
+	cases := map[string]struct {
+		input    string
+		expected []string
+	}{"Test1": {"abac", []string{"a2c"}},
+		"Test2": {"", []string{""}},
+		"Test3": {"a", []string{"a"}},
+	}
+
+	for name, test := range cases {
+		actual := numeronyms(test.input)
+		if !reflect.DeepEqual(actual, test.expected) {
+			t.Errorf(" test:%v input:%v, expected: %v, got: %v", name, test.input, test.expected, actual)
+		}
+	}
+
+}
+func TestLettersMulti(t *testing.T) {
```

Comment:
There is no need to have this split across multiple tests. The underlying test code is the same, it is only the cases which differ. You should add these cases into the above test and rename accordingly -`TestNumeronyms()` Can you please update this?

If you want each output to appear as an individual test then you can run them as sub tests within the one function. Like below:

```go

for name, test := range cases {
    input := test.input
    expected := test.expected
    t.Run(name, func(t *testing.T) {
        assert.Equal(t, expected, numeronyms(input), "Expected %v, Got %v", expected, numeronyms(input))
    })
}

```